ext/rugged/rugged_remote.c

```diff
@@ -620,6 +620,49 @@ static VALUE rb_git_remote_push(int argc, VALUE *argv, VALUE self)
 	return payload.result;
 }
 
+/*
+ *  call-seq:
+ *    remote.prune(options = {}) -> nil
+ *
+ *  Prune tracking refs that are no longer present on remote
+ *
+ *  Returns nil
+ *
+ *  The following options can be passed in the +options+ Hash:
+ *
+ *  :credentials ::
+ *    The credentials to use for the fetch operation. Can be either an instance of one
+ *    of the Rugged::Credentials types, or a proc returning one of the former.
+ *    The proc will be called with the +url+, the +username+ from the url (if applicable) and
+ *    a list of applicable credential types.
+ *
+ *  Example:
+ *
+ *    remote = Rugged::Remote.lookup(@repo, 'origin')
+ *    remote.prune()
+ *
+ */
+static VALUE rb_git_remote_prune(int argc, VALUE *argv, VALUE self)
+{
+	VALUE rb_options;
+	git_remote *remote;
+	git_fetch_options opts = GIT_FETCH_OPTIONS_INIT;
+	struct rugged_remote_cb_payload payload = { Qnil, Qnil, Qnil, Qnil, Qnil, rb_hash_new(), 0 };
+	int error;
+
+	Data_Get_Struct(self, git_remote, remote);
+	rb_scan_args(argc, argv, "01", &rb_options);
+	rugged_remote_init_callbacks_and_payload_from_options(rb_options, &opts.callbacks, &payload);
+
+	if ((error = git_remote_connect(remote, GIT_DIRECTION_FETCH, &opts.callbacks)) ||
+			(error = git_remote_prune(remote, &opts.callbacks)))
+		rugged_exception_check(error);
+
+	git_remote_disconnect(remote);
```

Comment:
Indeed, that's because it's an optional step during a fetch. The option to `#fetch` would be a tristate representing the field in the fetch optoins https://github.com/libgit2/libgit2/blob/master/include/git2/remote.h#L475-L488